### ISSUE TITLE
Syncing project fails

### ISSUE THREAD

**Issue by ALinuxPerson** [intellij-rust/intellij-rust#6603]
## Environment

* **IntelliJ Rust plugin version:** 0.3.138.3572-203
* **Rust toolchain version:** rustc 1.49.0 (e1884a8e3 2020-12-29)
* **IDE name and version:** CLion 2020.3.1
* **Operating system:** Arch Linux

## Problem description

Whenever I try to do any action that involves syncing the project like attaching a cargo project and opening a project, the build window appears stating that the cargo project sync has failed. This problem makes the IDE unusable as auto completion no longer works. This first appeared when I updated rust using `rustup update stable`. The logs can be found [here](https://github.com/intellij-rust/intellij-rust/files/5760310/idea.log), the specific error can be found at the bottom of the text.

I've tried,

- Invalidating caches then restarting
- Closing and reopening the IDE

![image](https://user-images.githubusercontent.com/55941755/103458557-470bc280-4d44-11eb-8a7b-7a6bd388904a.png)
Here is the build window I was talking about.

In short,

- Syncing a project fails
- In the logs, an error about a package missing appears
- Auto completion fails, making the IDE unusable

## Steps to reproduce

1. Update rust to 1.49.0.
2. Open a project or do anything that syncs the project.

**Comment by Undin** [comment]
@ALinuxPerson thanks for the report!

Could you archive your current `/home/alp/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/` directory and attach the archive to this issue?

As I can see, it's because of some bug during fetching actual stdlib metadata after toolchain update.
I see two workarounds:
- remove `/home/alp/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/vendor/` directory and try to sync cargo projects again. Not 100% sure that it helps
- disable `org.rust.cargo.fetch.actual.stdlib.metadata` experimental feature (`Help | Find Action`, type `Experimental feature` and disable the corresponding checkbox in the dialog). It should help anyway but prevents completion and name resolution for some stdlib modules (mostly for `std::os` module)

**Comment by ALinuxPerson** [comment]
Thanks! your first solution solved the problem for me. I've still provided the archive just in case. I'll be closing this issue now.

[rust.zip](https://github.com/intellij-rust/intellij-rust/files/5760382/rust.zip)

**Comment by Undin** [comment]
> I've still provided the archive just in case.

Thank you!

> I'll be closing this issue now.

Let's keep this issue open until it's fixed in the plugin

**Comment by kubijo** [comment]
Is there any progress on this?
I have this issue as well, and the proposed "workaround" doesn't do anything for me.

**Comment by vn971** [comment]
Hi, I have the same with `rustc 1.55.0 (c8dfcfe04 2021-09-06)`, `rustup 1.24.3 (2021-06-08)`,
```
IntelliJ IDEA 2021.2.2 (Community Edition)
Build #IC-212.5284.40, built on September 15, 2021
Runtime version: 11.0.12+7 amd64
VM: OpenJDK 64-Bit Server VM by Oracle Corporation
Linux 5.14.6-arch1-1
GC: ParNew, ConcurrentMarkSweep
Memory: 725M
Cores: 8
Non-Bundled Plugins: Dart (212.5284.31), ms.konovalov.intellij.hidpi-profiles (2017.1.3), org.sylfra.idea.plugins.linessorter (1.0.1), org.toml.lang (0.2.155.4114-212), org.rust.lang (0.4.155.4114-212)
Kotlin: 212-1.5.10-release-896-IJ5284.40
Current Desktop: XFCE
```

The work-around of disabling `org.rust.cargo.fetch.actual.stdlib.metadata` IDEA experimental feature (see above) did successfully work for me, although I'm worried about the long-levity of this work-around

**Comment by Undin** [comment]
@vn971 Could you provide logs to find out the cause, please? See [instruction](https://intellij-support.jetbrains.com/hc/en-us/articles/207241085-Locating-IDE-log-files) on how to find them

**Comment by vn971** [comment]
@Undin oh sure. I forgot to write initially, but the computer is also frequently in offline mode. All the Rust packages are already pre-downloaded though (I did it with `cargo metadata --format-version 1 --all-features`). With the feature being disabled, everything works perfectly, also while offline. Will attach the log a bit later, now gtg.

**Comment by Undin** [comment]
@vn971 Is it true that `Sync` fails only in offline mode? If yes, does the error look like
<img width="985" alt="Screen Shot 2021-09-22 at 16 48 51" src="https://user-images.githubusercontent.com/2539310/134356263-6b9d9724-c036-4e37-89c3-e82ccaaa7db7.png">
?
